### PR TITLE
[Close #199] improve protocol for deps

### LINKED ISSUE #199: Improve git_for_deps_erlang_mk and git_for_deps_rebar
We should allow the user to pick the regular expression they want to be validated over the repo urls instead of forcing ``git://``

### PR DESCRIPTION
**Pull request by Euen**
[Close #199] improve protocol for deps

### CODE REVIEW

File: src/elvis_project.erl [hunk 1]
```diff
@@ -14,8 +14,8 @@
         "commit.").
 
 -define(DEP_NO_GIT,
-        "Dependency '~s' is not using git protocol, "
-        "please change this to something like `git://...`").
+        "Dependency '~s' is not using correct protocol, "
```
Comment (elbrujohalcon) [inline src/elvis_project.erl]: correct -> appropriate

File: src/elvis_project.erl [hunk 2]
```diff
@@ -130,15 +134,21 @@ is_rebar_master_dep({_AppName, _Vsn, {_SCM, _Location, {branch, "master"}}}) ->
 is_rebar_master_dep(_) ->
     false.
 
-is_rebar_not_git_dep({_AppName, _Vsn, {_SCM, "git://" ++ _, _Branch}}) -> false;
-is_rebar_not_git_dep(_) -> true.
+is_rebar_not_git_dep({_AppName, _Vsn, {_SCM, Url, _Branch}}, Regex) ->
+    case re:run(Url, Regex, []) of
```
Comment (elbrujohalcon) [inline src/elvis_project.erl]: you can do just
```erlang
nomatch == re:run(Url, Regex, [])
```

File: src/elvis_project.erl [hunk 3]
```diff
@@ -150,16 +160,15 @@ is_erlang_mk_master_dep(Line) ->
...
-    [Protocol, Url | _] =
+    [_Protocol, Url | _] =
         [Part
             || Part <- binary:split(Dependency, <<" ">>, [global, trim])
              , Part /= <<>>],
-    case {Protocol, Url} of
-        {<<"git">>, <<"git://", _/binary>>} -> false;
-        {<<"git">>, _} -> true;
-        {_NotGit, _} -> false
+    case re:run(Url, Regex, []) of
```
Comment (elbrujohalcon) [inline src/elvis_project.erl]: you can do just
```erlang
nomatch == re:run(Url, Regex, [])
```

File: src/elvis_project.erl (hunk 2 above)
Comment (Euen) [inline src/elvis_project.erl]: :+1:  beautiful solution

**Pull request merged**